### PR TITLE
SOLID principles

### PR DESCRIPTION
**Pull request by olivierschultz**
SOLID principles

### CODE REVIEW

**Comment by olivierschultz** [comment]
@BerthelotAnthony Can you check and merge this one aswell pls?

**Comment by BerthelotAnthony** [comment]
@olivierschultz Of source Olivier, it will be my pleasure !

File: software-development-strategy/solid.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Software Development Strategy - SOLID
+
+__SOLID__ principles are fundamental when developing software. However, nowodays they can be applied to another process in software development lifecycle like QA Automation, CI/CD pipelines, Infrastructure Development etc.
```
Comment (BerthelotAnthony) [inline software-development-strategy/solid.md]: software**s** ?

File: software-development-strategy/solid.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+__SOLID__ is the abbreviation of the following core principles;
```
Comment (BerthelotAnthony) [inline software-development-strategy/solid.md]: why **;** ? Use **:**

File: software-development-strategy/solid.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+* __S__ - Single Responsability. It dictates that a class or method should have one responsability.
+* __O__ - Open/Closes. It didactes that a class or method should be extensible but shouldn't be modifiable.
+* __L__ - Liskov Substition. It didactes that subclasses should be exchangeable in between them without having any side effects.
+* __I__ - Interface Segregation. It didactes that a subclass shouldn't depend on the methods that it doesn't use. If there is such a method on the interface, it should be better to split that method into a different interface. 
+* __D__ - Dependency Inversion. It didactes that entities should depend on abstractions.
```
Comment (BerthelotAnthony) [inline software-development-strategy/solid.md]: **dictates**

File: software-development-strategy/solid.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Software Development Strategy - SOLID
+
+__SOLID__ principles are fundamental when developing software. However, nowodays they can be applied to another process in software development lifecycle like QA Automation, CI/CD pipelines, Infrastructure Development etc.
+
+__SOLID__ is the abbreviation of the following core principles;
+* __S__ - Single Responsability. It dictates that a class or method should have one responsability.
+* __O__ - Open/Closes. It didactes that a class or method should be extensible but shouldn't be modifiable.
+* __L__ - Liskov Substition. It didactes that subclasses should be exchangeable in between them without having any side effects.
+* __I__ - Interface Segregation. It didactes that a subclass shouldn't depend on the methods that it doesn't use. If there is such a method on the interface, it should be better to split that method into a different interface.
```
Comment (BerthelotAnthony) [inline software-development-strategy/solid.md]: **didactes**

File: software-development-strategy/solid.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Software Development Strategy - SOLID
+
+__SOLID__ principles are fundamental when developing software. However, nowodays they can be applied to another process in software development lifecycle like QA Automation, CI/CD pipelines, Infrastructure Development etc.
+
+__SOLID__ is the abbreviation of the following core principles;
+* __S__ - Single Responsability. It dictates that a class or method should have one responsability.
+* __O__ - Open/Closes. It didactes that a class or method should be extensible but shouldn't be modifiable.
+* __L__ - Liskov Substition. It didactes that subclasses should be exchangeable in between them without having any side effects.
```
Comment (BerthelotAnthony) [inline software-development-strategy/solid.md]: **didactes**

File: software-development-strategy/solid.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Software Development Strategy - SOLID
+
+__SOLID__ principles are fundamental when developing software. However, nowodays they can be applied to another process in software development lifecycle like QA Automation, CI/CD pipelines, Infrastructure Development etc.
+
+__SOLID__ is the abbreviation of the following core principles;
+* __S__ - Single Responsability. It dictates that a class or method should have one responsability.
+* __O__ - Open/Closes. It didactes that a class or method should be extensible but shouldn't be modifiable.
```
Comment (BerthelotAnthony) [inline software-development-strategy/solid.md]: **didactes**

**Pull request merged**